src/main/java/Storage.java

```diff
@@ -0,0 +1,98 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Storage {
+
+    private final String filePath;
+
+    Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads the tasks from the local storage file and creates a list of Tasks.
+     *
+     * @return a list of Tasks stored in the local storage file.
+     * @throws DukeException if the local storage file cannot be created or loaded.
+     */
+    ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> taskList = new ArrayList<>();
+
+        try {
+            //noinspection ResultOfMethodCallIgnored
+            new File(filePath).createNewFile();
+        } catch (IOException e) {
+            throw new DukeException("☹ OOPS!!! There was an error creating a save file.");
+        }
+
+        try {
+            Scanner fileReader = new Scanner(new File(filePath));
+            while (fileReader.hasNextLine()) {
+                String s = fileReader.nextLine();
+                Task task = getTask(s);
+                taskList.add(task);
+            }
+            return taskList;
+        } catch (FileNotFoundException | DukeException e) {
+            throw new DukeException("☹ OOPS!!! There was an error while loading the file");
+        }
+    }
+
+    /**
+     * Converts a string into a Task.
+     *
+     * @param s string description of a Task.
+     * @return a Task described by the input string.
+     * @throws DukeException if input string is in the wrong format.
+     */
+    private Task getTask(String s) throws DukeException {
```

Comment:
I find that 's' is not quite descriptive for a method argument name. I would suggest using "description" instead to make it more meaningful.